client/src/components/new-request/analyze-results.vue

```diff
@@ -270,13 +301,115 @@ export default class AnalyzeResults extends Vue {
     }
     return []
   }
+  get designations () {
+    if (this.issue && this.issue.designations) {
+      return this.issue.designations
+    }
+    return null
+  }
   get designationIsFixed () {
-    if (this.json.issues.every(issue => !newReqModule.designationIssueTypes.includes(issue.issue_type))) {
-      if (this.isLastIndex) {
-        return true
+    try {
+      if (this.userCancelled) return false
+      if (!this.changesInBaseName) {
+        // creating a new version of allDesignationsList because we are going to modify it but still need the original
+        let AllDesignationsList = allDesignationsList
+        // allDesignations is an array of objects containing {words, end} keys, set designationEntityTypes to the
+        // relevant object
+        const designationEntityTypes = allDesignations[this.entity_type_cd]
+        /* designationEntityTypes.words.length === 0 is true for types 'PAR', 'FI', 'PA', and the proprietorships so
+         the designation issue is fixed as long as there are no designations present and no nameAction words which in
+         this situation would only be designation-like words to be removed */
+        if (designationEntityTypes && designationEntityTypes.words.length === 0) {
```

Comment:
FYI, you can now do `if (designationEntityTypes?.words.length === 0)...`.

If "designationEntityTypes" is falsey they the statement is undefined (which is not === 0).